### PR TITLE
Fuzz ssz round trip

### PR DESCRIPTION
**Pull request by bshastry**
**What type of PR is this?**

Other (adds fuzzer)

**What does this PR do? Why is it needed?**

This PR adds a new fuzzer for testing ssz roundtrip coding. It is needed to increase test coverage.

**Which issues(s) does this PR fix?**

None

**Other notes for review**

Incorporates earlier feedback from private PR.

### CODE REVIEW

File: beacon-chain/p2p/encoder/ssz_test.go [hunk 1]
```diff
@@ -9,15 +9,571 @@ import (
...
+	assertProtoMessagesEqual(t, decoded, msg)
+}
+
+func FuzzRoundTripWithGossip(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte, index int) {
+		if index < 0 {
+			t.Skip()
+		}
+		// Select a random creator from the list.
+		creator := creators[index%len(creators)]
```
Comment (prestonvanloon) [inline beacon-chain/p2p/encoder/ssz_test.go:526]: Using `index%len(creators)` will invalidate the test corpus if any new creator is added.

```suggestion
		if index < 0 || index >= len(creators) {
			t.Skip()
		}
		// Select a random creator from the list.
		creator := creators[index]
```

File: beacon-chain/p2p/encoder/ssz_test.go [hunk 2]
```diff
@@ -9,15 +9,571 @@ import (
...
+
+	data := []byte("\xc800")
+	msg := &ethpb.Consolidation{}
+
+	if err := proto.Unmarshal(data, msg); err != nil {
+		t.Logf("Failed to unmarshal: %v", err)
+		return
+	}
+
+	fmt.Printf("msg=%#+v\n", msg)
```
Comment (prestonvanloon) [inline beacon-chain/p2p/encoder/ssz_test.go:568]:
```suggestion
```

File: beacon-chain/p2p/encoder/ssz_test.go (hunk 1 above)
Comment (bshastry) [reply beacon-chain/p2p/encoder/ssz_test.go]: Done

**Comment by prestonvanloon** [comment]
Please run `bazel run //:gazelle`

File: beacon-chain/p2p/encoder/ssz_test.go (hunk 2 above)
Comment (bshastry) [reply beacon-chain/p2p/encoder/ssz_test.go]: Done

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**